packages/wagmi/src/connectors/EmailConnector.ts

```diff
@@ -0,0 +1,103 @@
+import type { Chain } from '@wagmi/core'
+import { Connector } from '@wagmi/core'
+import { W3mFrameProvider } from '@web3modal/wallet'
+import { createWalletClient, custom, SwitchChainError } from 'viem'
+
+interface W3mFrameProviderOptions {
+  projectId: string
+}
+
+export class EmailConnector extends Connector<W3mFrameProvider, W3mFrameProviderOptions> {
+  readonly id = 'w3mEmail'
+
+  readonly name = 'Web3Modal Email'
+
+  readonly ready = true
+
+  private provider: W3mFrameProvider = {} as W3mFrameProvider
+
+  public constructor(config: { chains?: Chain[]; options: W3mFrameProviderOptions }) {
+    super(config)
+    if (typeof window !== 'undefined') {
```

Comment:
How would this play out in Next.js apps and RSCs ?

Reply:
Web3Modal is only usable on client so it will be behind `use client` component. We actually have this set up for secure.web3modal.com